app/components/maps/SidebarCharts.tsx

```diff
@@ -0,0 +1,385 @@
+    }
+
+    // give more space when > 1 tooltip per replay
+    if (metric.chartDataCallback.length > 1) {
+        options.chart.height += replaysData.length * 34;
+    }
+
+    options.title.text = metric.name;
+    options.series = replaySeries;
+    options.xAxis.events = {
+        afterSetExtremes(event: any) {
+            callBack({
+                Event: event,
+                Metric: metric,
+            });
+        },
+    };
+    options.plotOptions = {
+        series: {
+            events: {
+                mouseOut: () => {
+                    globalChartsData.hoveredRaceTime = undefined;
+                },
+            },
+            point: {
+                events: {
+                    mouseOver: (event: any) => {
+                        event.preventDefault();
+                        if (!event.target.isNull && typeof event.target.x === 'number') {
+                            globalChartsData.hoveredRaceTime = event.target.x;
+                        }
+                    },
+                },
+            },
+        },
+    };
+
+    const highCharts = <HighchartsReact constructorType="stockChart" highcharts={Highcharts} options={options} />;
+
+    clearInterval(globalInterval);
+
+    if (syncWithTimeLine) {
+        globalInterval = setInterval(() => {
+            if (timeLineGlobal.isPlaying || prevCurrentRacetime !== timeLineGlobal.currentRaceTime) {
+                prevCurrentRacetime = timeLineGlobal.currentRaceTime;
+                let validCharts: boolean = false;
+                highCharts.props.highcharts.charts.forEach((chart: any) => {
+                    if (chart) {
+                        validCharts = true;
+                        const matchingPts: any[] = [];
+                        chart.series.forEach((serie: any) => {
+                            for (let i = 0; i < serie.points.length; i++) {
+                                if (serie.points[i].category >= timeLineGlobal.currentRaceTime) {
+                                    matchingPts.push(serie.points[i]);
+                                    serie.points[i].setState('hover');
+                                    break;
+                                }
+                            }
+                        });
+                        if (matchingPts.length) {
+                            chart.tooltip.refresh(matchingPts);
+                        }
+                    }
+                });
+                if (!validCharts) {
+                    clearInterval(globalInterval);
+                }
+            }
+        }, 1);
+    } else {
+        clearInterval(globalInterval);
+    }
+
+    addChartFunc(highCharts);
+    return (
+        highCharts
+    );
+};
+
+// eslint-disable-next-line no-undef
+let timer: NodeJS.Timeout;
+
+const debounce = (callback: () => any, timeout: number) => () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+        callback();
+    }, timeout);
+};
```

Comment:
I've seen this debounce method in some different places, we might want to extract this to some util file